src/Render/TagHooks.php

```diff
@@ -98,9 +123,41 @@ public function renderForm( $input, array $args, Parser $parser, PPFrame $frame
 			];
 		}
 
+		if ( isset( $_COOKIE['ffSaveFields'] ) ) {
+			Core::addPreSaved( json_decode( base64_decode( $_COOKIE['ffSaveFields'] ), true ) );
+			setcookie(
+				"ffSaveFields",
+				"",
+				time() - 3600,
+				'/'
+			);
+		}
+
 		Core::$securityId = uniqid();
 		Core::$chkSums = [];
 		Core::includeTagsCSS( Core::getRealUrl() . '/Modules/ext.WSForm.css' );
+
+		// Are there explicit 'restrictions' lifts set?
+		// TODO: Allow administrators of a wiki to configure whether lifting restrictions is allowed (useful for public wikis)
+		if ( isset( $args['restrictions'] ) ) {
+			// Parse the given restriction
+			$restrictions = $parser->recursiveTagParse(
+				$args['restrictions'],
+				$frame
+			);
+
+			// Only allow anonymous users if the restrictions are lifted
+			$allowAnonymous = strtolower( $restrictions ) === 'lifted';
+
+			unset( $args['restrictions'] );
+		} else {
+			// By default, deny anonymous users
+			$allowAnonymous = false;
+		}
+
+		if ( !$this->officialForm && $allowAnonymous === false ) {
```

Comment:
What if I make a form with `restrictions="lifted"`? Would it then be rendered regardless of wether it is approved?